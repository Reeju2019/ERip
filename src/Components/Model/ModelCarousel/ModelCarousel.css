.service-img {
  max-width: 2vw;
}

.images {
  width: 8vmax;
  height: 8vmax;
}

.service-para {
  font-size: 1vmax;
  display: flex;
  justify-content: center;
}
.caro{
  background-color: #f5f6fa;
}