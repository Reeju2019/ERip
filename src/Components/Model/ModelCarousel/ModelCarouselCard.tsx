import * as React from 'react'

interface IModelCarouselCardProps {
  data: {
    model: string
    image: string
  }
}

const ModelCarouselCard: React.FunctionComponent<IModelCarouselCardProps> = ({ data }) => {
  return (
    <>
      <div className='d-flex flex-column m-2'>
        <div className='p-2 bg-white'>
          <div className='service-img'>
            <img
              src={data.image}
              className='images d-flex justify-content-center'
              alt={data.model}
            />
          </div>
        </div>
        <p className='service-para'>{data.model}</p>
      </div>
    </>
  )
}

export default ModelCarouselCard
