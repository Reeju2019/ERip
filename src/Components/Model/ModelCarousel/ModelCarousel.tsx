import React from 'react'
import { Container, Row } from 'react-bootstrap'
import AllModelsData from '../../../Data/AllModels.mock.json'
import ModelCarouselCard from './ModelCarouselCard'
import './ModelCarousel.css'
import Carousel from 'react-multi-carousel'
import 'react-multi-carousel/lib/styles.css'

const ModelCarousel: React.FunctionComponent = () => {
  const responsive = {
    desktop: {
      breakpoint: { max: 3000, min: 1024 },
      items: 4,
      slidesToSlide: 1,
    },
    tablet: {
      breakpoint: { max: 1024, min: 464 },
      items: 2,
      slidesToSlide: 1,
    },
    mobile: {
      breakpoint: { max: 464, min: 0 },
      items: 2,
      slidesToSlide: 1,
    },
  }
  return (
    <>
      <section className='caro'>
        <Container>
          <Row>
            <h4 className='d-flex justify-content-center mt-5'>Top Required Models</h4>
          </Row>
          <Row>
            <div className='col-12 service-content mb-5'>
              <Carousel
                swipeable={true}
                draggable={true}
                showDots={true}
                responsive={responsive}
                ssr={true}
                infinite={true}
                autoPlay={false}
                autoPlaySpeed={1000}
                keyBoardControl={true}
                customTransition='all .5'
                transitionDuration={500}
                containerClass='carousel-container'
                removeArrowOnDeviceType={['tablet', 'mobile']}
                dotListClass='custom-dot-list-style'
                itemClass='carousel-item-padding-40-px'
                partialVisible={true}
                rewindWithAnimation={true}
                sliderClass='react-multi-carousel-track'
              >
                {AllModelsData.iPhone.map((item, index) => {
                  return (
                    <div key={index} className='carousel-item'>
                      <ModelCarouselCard data={item} />
                    </div>
                  )
                })}
              </Carousel>
            </div>
          </Row>
        </Container>
      </section>
    </>
  )
}

export default ModelCarousel
