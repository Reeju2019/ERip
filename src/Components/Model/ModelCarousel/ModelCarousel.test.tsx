import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import ModelCarousel from './ModelCarousel'


test('Model carousel', () => {
    render(<ModelCarousel />)
})
describe('Model carousel rendering specification', () => {
    it('SnapShot testing', () => {
        const component = renderer.create(<ModelCarousel />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('render without crashing Model carousel', () => {
        const div = document.createElement('div')
        render(<ModelCarousel />)
        ReactDOM.unmountComponentAtNode(div)
    })
})
