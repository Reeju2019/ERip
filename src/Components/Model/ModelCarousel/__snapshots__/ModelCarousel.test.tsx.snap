// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model carousel rendering specification SnapShot testing 1`] = `
<section
  className="caro"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <h4
        className="d-flex justify-content-center mt-5"
      >
        Top Required Models
      </h4>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 service-content mb-5"
      >
        <div
          className="react-multi-carousel-list carousel-container "
          dir="ltr"
        >
          <ul
            className="react-multi-carousel-track react-multi-carousel-track"
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "overflow": "unset",
                "transform": "translate3d(0px,0,0)",
                "transition": "none",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</section>
`;
